shared/lib_shared_inverter.cpp

```diff
@@ -8,9 +9,93 @@ SharedInverter::SharedInverter(int inverterType, int numberOfInverters,
 	m_numInverters = numberOfInverters;
 	m_sandiaInverter = sandiaInverter;
 	m_partloadInverter = partloadInverter;
+	m_tempEnabled = false;
 }
 
-void SharedInverter::calculateACPower(const double powerDC_Watts, const double DCStringVoltage)
+bool sortByVoltage(const double i[], const double j[])
+{
+	return (i[0] < j[0]);
+}
+
+bool SharedInverter::setTempDerateCurves(double* curve1, double* curve2, double* curve3)
+{
+	if (curve1[0] < 0 || curve1[1] <= -99 || curve1[2] >= 0) return false;

```

Comment:
Where do these limits come from? Worth a comment. Also possibly worth creating variables for the limits if they're going to apply to multiple derate curves, so that if we want to change the limits, we only have to change them in one place.

Reply:
Good question. The first comes from the fact that the voltage needs to be above 0 in order for any derating to be applied. The third comes from the fact that the derating needs to be negative. The second one I changed to be <=-273 since that's the lowest possible temperature in C :)